pkgs/development/compilers/dmd/sourcebuild.nix

```diff
@@ -0,0 +1,173 @@
+{ stdenv, lib, fetchFromGitHub
+, makeWrapper, unzip, which, writeTextFile
+, curl, tzdata, gdb, darwin, git, callPackage
+, targetPackages, fetchpatch, bash
+, HOST_DMD
+, version
+, dmdSha256
+, druntimeSha256
+, phobosSha256
+}:
+
+let
+
+  dmdConfFile = writeTextFile {
+      name = "dmd.conf";
+      text = (lib.generators.toINI {} {
+        Environment = {
+          DFLAGS = ''-I@out@/include/dmd -L-L@out@/lib -fPIC ${lib.optionalString (!targetPackages.stdenv.cc.isClang) "-L--export-dynamic"}'';
+        };
+      });
+  };
```

Comment:
4 spaces intendation does not follow our editorconfig. Please format it with nixpkgs-fmt.